### PR TITLE
feat:[CDS-49121]:added-elastigroup-resource-text

### PR DESCRIPTION
**Pull request by namanharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx [hunk 1]
```diff
@@ -51,9 +51,16 @@ export default function ResourceConstraintTooltip(props: ResourceConstraintToolt
     props?.data?.executionList,
     props.data?.executionId
   )
+  const getDeploymentTypeResourceWaitingText = (deploymentType: string): string => {
+    if (isElastigroupDeploymentType(deploymentType))
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx:55]: Add curly braces for better readability.
`if (condition) {}`

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx [hunk 2]
```diff
@@ -51,9 +51,16 @@ export default function ResourceConstraintTooltip(props: ResourceConstraintToolt
...
+      return getString('pipeline.resourceConstraints.elastigroupInfraEntity')
+    else if (isServerlessDeploymentType(deploymentType))
+      return getString('pipeline.resourceConstraints.serverlessInfraEntity')
 
+    return getString('pipeline.resourceConstraints.k8sNamespaceText')
+  }
   const stageDeploymentType = module ? props.data?.moduleInfo?.[module]?.serviceInfo?.deploymentType : undefined
-  const isServerlessDeploymentTypeSelected = isServerlessDeploymentType(defaultTo(stageDeploymentType, ''))
+  const getDeploymentTypeSelected = getDeploymentTypeResourceWaitingText(stageDeploymentType as string)
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx:63]: Change to better variable name. Ex - `resourceWaitingText`

Comment (namanharness) [reply src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx]: done

File: src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx (hunk 1 above)
Comment (namanharness) [reply src/modules/70-pipeline/pages/execution/ExecutionPipelineView/ExecutionGraphView/ExecutionStageDetails/components/ResourceConstraints/ResourceConstraints.tsx]: resolving with switch case as discussed

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**